### ISSUE TITLE
[BITV] 9.4.1.2/2.2 - The "Weather" menu (see screenhot) has been implemented using the aria-haspopup="menu" attribute, which informs users that a menu is being opened.(1)

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#41889]
However, even though the menu items can be navigated using arrow keys, the corresponding ARIA roles role="menu" and role="menuitem" are completely missing. 

![image](https://github.com/nextcloud/server/assets/6078378/82e14a9a-9359-4e62-bf9a-9667882c3623)

https://report.bitvtest.de/default-en/cbedb5ef-4815-4cba-a83d-995599c178c3.html#checkpoint-bfbf07e5c3-v2-n2

**Comment by ShGKme** [comment]
`aria-haspopup` is to be fixed by https://github.com/nextcloud-libraries/nextcloud-vue/pull/4887

However, "the corresponding ARIA roles role="menu" and role="menuitem" are completely missing" conflicts with the previous issue: https://github.com/nextcloud/server/issues/37099#issuecomment-1785203065
